### PR TITLE
export max ID of tokenPair

### PR DESCRIPTION
**Pull request by ylsGit**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/dex/keeper/expected_keeper.go [hunk 1]
```diff
@@ -54,6 +54,8 @@ type IKeeper interface {
 	CompleteWithdraw(ctx sdk.Context, addr sdk.AccAddress) error
 	IterateWithdrawInfo(ctx sdk.Context, fn func(index int64, withdrawInfo types.WithdrawInfo) (stop bool))
 	DeleteWithdrawCompleteTimeAddress(ctx sdk.Context, timestamp time.Time, delAddr sdk.AccAddress)
+	GetTokenPairMaxID(ctx sdk.Context) (tokenPairMaxID uint64)
+	SetTokenPairMaxID(ctx sdk.Context, tokenPairMaxID uint64)
```
Comment (ilovers) [inline x/dex/keeper/expected_keeper.go:58]: I think MaxTokenPairID is better than TokenPairMaxID, and should keep naming consistent:
GetTokenPairMaxID-> GetMaxTokenPairID
SetTokenPairMaxID-> SetMaxTokenPairID

File: x/dex/keeper/keeper.go [hunk 2]
```diff
@@ -79,7 +79,7 @@ func (k Keeper) deleteUserTokenPair(ctx sdk.Context, owner sdk.AccAddress, pair
 func (k Keeper) SaveTokenPair(ctx sdk.Context, tokenPair *types.TokenPair) error {
 	store := ctx.KVStore(k.tokenPairStoreKey)
 
-	tokenPairNumber := k.GetTokenPairNum(ctx)
```
Comment (ilovers) [inline x/dex/keeper/keeper.go:82]: It is recommended to rename tokenPairNumber to maxTokenPairID

File: x/dex/keeper/keeper.go [hunk 3]
```diff
@@ -79,7 +79,7 @@ func (k Keeper) deleteUserTokenPair(ctx sdk.Context, owner sdk.AccAddress, pair
...
+	tokenPairNumber := k.GetTokenPairMaxID(ctx)
 	// list new tokenPair
 	if tokenPair.ID == 0 {
 		tokenPair.ID = tokenPairNumber + 1
```
Comment (ilovers) [inline x/dex/keeper/keeper.go:85]: ### suggestion
Rename tokenPairNumber to maxTokenPairID

File: x/dex/keeper/keeper.go [hunk 4]
```diff
@@ -452,16 +450,23 @@ func (k *Keeper) SetGovKeeper(gk GovKeeper) {
...
-	b := store.Get(types.TokenPairNumberKey)
+	b := store.Get(types.TokenPairMaxIDKey)
 	if b != nil {
-		k.cdc.MustUnmarshalBinaryBare(b, &tokenPairNumber)
+		k.cdc.MustUnmarshalBinaryBare(b, &tokenPairMaxID)
 	}
 	return
 }
 
+// SetTokenPairNum sets the max ID of token pair
```
Comment (ilovers) [inline x/dex/keeper/keeper.go:463]: Wrong func name(SetTokenPairNum) in comment.

File: x/dex/types/keys.go [hunk 5]
```diff
@@ -40,8 +40,8 @@ var (
 
 	// TokenPairKey is the store key for token pair
 	TokenPairKey = []byte{0x01}
-	// TokenPairNumberKey is the store key for token pair num
-	TokenPairNumberKey = []byte{0x02}
+	// TokenPairMaxIDKey is the store key for token pair max ID
+	TokenPairMaxIDKey = []byte{0x02}
```
Comment (ilovers) [inline x/dex/types/keys.go:44]: I think MaxTokenPairID is better than TokenPairMaxID, and should keep naming consistent:
TokenPairMaxIDKey -> MaxTokenPairIDKey

File: x/dex/keeper/keeper.go [hunk 6]
```diff
@@ -452,16 +450,23 @@ func (k *Keeper) SetGovKeeper(gk GovKeeper) {
...
+// GetMaxTokenPairID returns the max ID of token pair
+func (k Keeper) GetMaxTokenPairID(ctx sdk.Context) (tokenPairMaxID uint64) {
 	store := ctx.KVStore(k.tokenPairStoreKey)
-	b := store.Get(types.TokenPairNumberKey)
+	b := store.Get(types.MaxTokenPairIDKey)
 	if b != nil {
-		k.cdc.MustUnmarshalBinaryBare(b, &tokenPairNumber)
+		k.cdc.MustUnmarshalBinaryBare(b, &tokenPairMaxID)
 	}
 	return
```
Comment (summerpro) [inline x/dex/keeper/keeper.go:460]: It's best to return an error

**Pull request merged**